src/intrinsic/annotations.rs

```diff
@@ -65,19 +71,57 @@ impl AnnotationError {
                         "There must not be any statements after this annotated statement (and the annotated statement must not be nested in a block).",
                     ))
             }
+            AnnotationError::CalculusEncodingMismatch(span, calculus_name, encoding_name ) => {
+                Diagnostic::new(ReportKind::Error, span)
+                    .with_message(format!(
+                        "The '{}' calculus does not support the '{}' encoding.",
+                        calculus_name.name, encoding_name.name
+                    ))
+                    .with_label(Label::new(span).with_message(
+                        "The calculus and encoding are incompatible.",
+                    ))
+            }
+            AnnotationError::UnknownAnnotation(span, anno_name ) => {
+                Diagnostic::new(ReportKind::Error, span)
+                    .with_message(format!(
+                        "The '{}' annotation is unknown.",
+                        anno_name.name
+                    ))
+                    .with_label(Label::new(span).with_message(
+                        "This annotation is not defined.",
+                    ))
+            }
         }
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct Calculus {
```

Comment:
I don't quite understand why this struct exists. Shouldn't the `CalculusType` enum be sufficient? Why have the additional `Ident`?